dhis-2/dhis-api/src/main/java/org/hisp/dhis/scheduling/JobRunErrorsParams.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright (c) 2004-2023, University of Oslo
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ * Redistributions of source code must retain the above copyright notice, this
+ * list of conditions and the following disclaimer.
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.scheduling;
+
+import java.util.Date;
+import java.util.List;
+import javax.annotation.CheckForNull;
+import lombok.Data;
+import lombok.experimental.Accessors;
+import org.hisp.dhis.common.OpenApi;
+import org.hisp.dhis.common.UID;
+import org.hisp.dhis.feedback.ErrorCode;
+import org.hisp.dhis.user.User;
+
+/**
+ * Query params when searching for {@link JobConfiguration}s with errors.
+ *
+ * <p>A match has to satisfy all filters (AND logic) but only one of the given codes or object
+ * {@link UID} (OR logic).
+ *
+ * <p>If any of the criteria is not defined it has no filter effect.
+ *
+ * @author Jan Bernitt
+ */
+@Data
+@Accessors(chain = true)
+public class JobRunErrorsParams {
+
+  @OpenApi.Ignore @CheckForNull private UID job;
+
+  /** The user that ran the job */
+  @OpenApi.Property({UID.class, User.class})
+  @CheckForNull
+  private UID user;
+
+  /** The earliest date the job ran that should be included */
+  @CheckForNull private Date from;
```

Comment:
Can we use `LocalDate` or `LocalDateTime` for new models seeing as we want to move to use these eventually? Or is there something restricting us from doing this?

Reply:
I think I should try at least, just did this out of a (bad) habit

Reply:
We don't have setup a flexible parsing for `LocalDate` in the same way we have for `Date` so for now I keep it `Date`. At some point we should make an attempt to switch dates in APIs over while supporting the parsing flexibility.